http3/server.go

```diff
@@ -127,17 +165,52 @@ func (s *Server) ListenAndServeTLS(certFile, keyFile string) error {
 	config := &tls.Config{
 		Certificates: certs,
 	}
-	return s.serveImpl(config, nil)
+	return s.serveConn(config, nil)
 }
 
 // Serve an existing UDP connection.
 // It is possible to reuse the same connection for outgoing connections.
 // Closing the server does not close the packet conn.
 func (s *Server) Serve(conn net.PacketConn) error {
-	return s.serveImpl(s.TLSConfig, conn)
+	return s.serveConn(s.TLSConfig, conn)
+}
+
+// Serve an existing QUIC listener.
+// Make sure you use http3.ConfigureTLSConfig to configure a tls.Config
+// and use it to construct a http3-friendly QUIC listener.
+// Closing the server does close the listener.
+func (s *Server) ServeListener(listener quic.EarlyListener) error {
+	return s.serveImpl(func() (quic.EarlyListener, error) { return listener, nil })
+}
+
+func (s *Server) serveConn(tlsConf *tls.Config, conn net.PacketConn) error {
+	return s.serveImpl(func() (quic.EarlyListener, error) {
```

Comment:
Is there any reason not to construct the `quic.EarlyListener` here, and pass it to `serveImpl` directly, instead of passing this massive constructor callback?

Reply:
I decided to use a callback here since we only construct the `quic.EarlyListener` after a couple of if's and other statements. Moving that code to a different method like `Server.preServe` which should be called here and in `Server.ServeListener` (basically everywhere before the actual `serveImpl`) also doesn't seem like the best way, at least to me. What do you think?

Reply:
Fair enough, let's keep it this way.